### ISSUE TITLE
Library build targets in Cargo.toml can make rustfmt hang

### ISSUE THREAD

**Issue by hsivonen** [rust-lang-nursery/rustfmt#828]
Steps to reproduce:

Clone [encoding-rs](https://github.com/hsivonen/encoding-rs/) and checkout the revision [a89396c1abe61a30176ddf27eb87f43ad3c8b053](newest as of filing this issue). Run `cargo fmt`. Watch the process not finishing in a reasonable time. Press ctrl-C to stop the process. Checkout the previous revision [1c1ae044b9c93c9a9fd3811ad2f7fe2001099f57](https://github.com/hsivonen/encoding-rs/commit/1c1ae044b9c93c9a9fd3811ad2f7fe2001099f57). Run `cargo fmt`. Watch the process finish.

The change that confuses rustfmt is:

```
diff --git a/Cargo.toml b/Cargo.toml
index 4ae558c..3fade1d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,5 +7,12 @@ readme = "README.md"
 repository = "https://github.com/hsivonen/encoding-rs"
 build = "build.rs"

+[lib]
+name = "encoding_rs"
+crate-type = ["rlib", "staticlib"]
+
 [build-dependencies]
-rusty-cheddar = "0.3.1"
\ No newline at end of file
+rusty-cheddar = "0.3.1"
+
+[profile.release]
+lto = true
\ No newline at end of file
```

**Comment by hsivonen** [comment]
Commenting out the `crate-type` line makes rustfmt not hang.

**Comment by hsivonen** [comment]
Happens with rustfmt 0.3.0 and rustfmt 0.2.1.

**Comment by lunaryorn** [comment]
I'm seeing this as well, with cargo 0.13 nightly at `109cb7c` and rustfmt 0.6.2

**Comment by nrc** [comment]
Should be fixed by #1238. Please let me know if not.
